elasticsearch.go

```diff
@@ -189,12 +218,107 @@ func NewClient(cfg Config) (*Client, error) {
 		go client.DiscoverNodes()
 	}
 
-	return client, nil
+	if cfg.UseHeaderCheckOnly {
+		useHeaderCheckOnly = true
+	}
+
+
+	return client, err
+}
+
+func genuineCheckHeader(header http.Header) error {
+	if header.Get("X-Elastic-Product") != "Elasticsearch" {
+		return fmt.Errorf(unknownProduct)
+	}
+	return nil
+}
+
+func genuineCheckInfo(info info) error {
+	major, minor, _, err := ParseElasticsearchVersion(info.Version.Number)
+	if err != nil {
+		return err
+	}
+
+	if major < 6 {
```

Comment:
This function might do well with some prose comments on what is checked for which version ranges?